cmd/krel/cmd/changelog.go

```diff
@@ -205,22 +204,18 @@ func generateReleaseNotes(branch, startRev, endRev string) (string, error) {
 		return "", err
 	}
 
-	// Create the GitHub API client
-	ctx := context.Background()
-	httpClient := oauth2.NewClient(ctx, oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: changelogOpts.token},
-	))
-	githubClient := github.NewClient(httpClient)
+	c, err := notesOptions.Client()
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get GitHub API client")
+	}
 
 	gatherer := &notes.Gatherer{
-		Client:  notes.WrapGithubClient(githubClient),
-		Context: ctx,
-		Org:     git.DefaultGithubOrg,
-		Repo:    git.DefaultGithubRepo,
+		Client:  c,
+		Context: context.Background(),
+		Options: notesOptions,
```

Comment:
I find it confusing that I pass the Client to the gatherer but also the options, which seem to be the canonical way on how to create a client. Why both?

Reply:
I changed the gatherer to have only private fields but two convenience `New...` functions. This makes the overall usage more straight forward.